git/git.go

```diff
@@ -21,27 +21,22 @@ func VerifyRef(ref string) bool {
 
 // CurrentBranch reads the checked-out branch for the git repository
 func CurrentBranch() (string, error) {
-	noBranchError := errors.New("git: not on any branch")
+	refCmd := GitCommand("symbolic-ref", "--quiet", "--short", "HEAD")
 
-	// we avoid using `git branch --show-current` for compatibility with git < 2.22
-	output, err := run.PrepareCmd(GitCommand("rev-parse", "--abbrev-ref", "HEAD")).Output()
-	branchName := firstLine(output)
+	output, err := run.PrepareCmd(refCmd).Output()
 	if err == nil {
-		if branchName == "HEAD" {
-			return "", noBranchError
-		} else {
-			return branchName, nil
-		}
+		// Found the branch name
+		return firstLine(output), nil
 	}
 
-	// Fall back to symbolic-ref in case we're in a repository with no commits
-	output, err = run.PrepareCmd(GitCommand("symbolic-ref", "--short", "HEAD")).Output()
-	if err != nil {
-		return "", noBranchError
+	ce := err.(*run.CmdError)
```

Comment:
This is going to panic if err is not of type `*run.CmdError`.

As a rule of thumb, we should never assume that an error is guaranteed to be of any specific type, but always check the type:
```go
var cmdErr *run.CmdError
if errors.As(err, &cmdErr) {
	cmdErr.Stderr.Len()
}
```

Reply:
hm; I opted for the panic because it seemed like err was _always_ supposed to be a `*run.CmdError` and panicking would signal to a developer that they were changing that contract. If we silently ignore the failed cast, doesn't that mean we can let a bug go unacknowledged?

Reply:
ah, re-reading more closely i see it's possible to not get a `CmdError` out of a prepared command. I'll fix the type assertion.